cmd/shfmt/main.go

```diff
@@ -322,16 +326,34 @@ func diffBytes(b1, b2 []byte, path string) ([]byte, error) {
 		// as diff will return non-zero if the files differ.
 		return nil, err
 	}
+
+	output := make([][]byte, 0)
```

Comment:
use a `bytes.Buffer` instead, and then the functions can use `fmt.Fprint(w, ...)`. this also means we won't need to use `bytes.Join`.